.travis.yml

```
@@ -75,10 +75,10 @@ jobs:
     - rvm: ruby-head
       env: RGV=master
       stage: test
-    # 1.x mode (we want to keep stuff passing in 1.x mode for now)
-    - rvm: 2.6.1
-      env: RGV=v2.7.8 BUNDLER_SPEC_SUB_VERSION=1.98
+    # 3.x mode
+    - rvm: 2.5.3
       stage: test
+      env: RGV=v3.0.2 BUNDLER_SPEC_SUB_VERSION=3.0.0
```

```suggestion
      env: RGV=v3.0.2 BUNDLER_SPEC_SUB_VERSION=2.98
```

I think we should instead bump the version to `3.0.0.dev` in the version file, and run the full matrix against that (with all the feature flags turned on). And only keep this extra entry testing the 2.x mode.

After all, the main point of master's CI is to test the quality of the latest master's code. This would not exercise that (except for this entry) because all feature flags would be off.

---

I'm not ignoring this. I'm writing a proposal that will talk about what i'm doing here and why. I'll let you know when i open a Github issue.

---

https://github.com/bundler/bundler/issues/6987

---

I'm going to be posting in here instead of making an entire proposal as that idea backfired on me.

I've been thinking about this over the past few days and i definitely agree that we should be testing Bundler 3 with all the feature flags/version checks enabled, but i don't want to bump the version to v3.0.0.

But seeing that Bundler 2 is the version that's currently being actively developed and most people are contributing to, i think that `Bundler::VERSION` should reflect that and stick with being `2.1.0.dev`. Once the release manager for Bundler 3 begins work on the first Bundler 3 pre-release, then the major version should be bumped.

What i think is a good in-between solution is to add more jobs to Travis to test more versions of Ruby and RubyGems to make sure we're not breaking any breaking feature changes for Bundler 3.

---

@colby-swandale Your reasoning looks sensible to me. :+1:

My proposal: implement my suggestion in #6956, and also add entries for Bundler 3 + MRI 2.4 and higher, and rubygems 3 and higher. From a quick and dirty calculation, that would leave our CI matrix with 17 entries versus the current 14, which doesn't sound too bad.

What do you think?

---

Personally, I would make a separae PR for #6956. Can you list which Bundler 3 entries should also be added?

---

Or just edit the PR.

---

Ok! I would add the following entries to run with `BUNDLER_SPEC_SUB_VERSION=3.0.0`

  MRI \ Rubygems | 3.0.2 | master
  ---------------------- | -------| ---------| 
  2.4.5 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
  2.5.3 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
  2.6.1 | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
  ruby-head (allowing failures) | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |

---

Oh yea, that's defs a good idea.

---

So i like the coverage but the thing the concerns me now is the CI time. Even just adding a few new records to the build matrix is kinda blowing out the time for CI to run. See https://travis-ci.org/bundler/bundler/jobs/499223906

Maybe we should limit the list to a subset that we consider essential?

---

He, well that's why I was suggesting to implement #6956, because it removes non essential entries and thus makes this change less noticiable (only 3 more entries from the current matrix I think). Maybe I propose #6956 separately, and then we rebase this on top of this and see how it goes?

---

yea, that sounds like a good plan. 👍

---

On it!